web/src/reducers/queue.js

```diff
@@ -1,51 +1,16 @@
 import {INITIALIZE, ADD_SONG, REMOVE_SONG, SET_LOCAL_QUEUE, SET_SONG_STATUS} from '../actions';
 
-let i = 0;
-
-function updateStatus(item, type, data) {
-  switch (type) {
-    case 'error':
-      return {...item, error: data, status: data.error};
-    case 'meta':
-      return {...item, song: data, status: 'waiting...'};
-    case 'downloading':
-      return {...item, status: 'downloading...'};
-    case 'processing':
-      return {...item, status: 'processing...'};
-    case 'done':
-      return {...item, status: null};
-    default:
-      console.log(type, data);
-      return {...item, status: 'unknown'};
-  }
-}
-
 export default function queue(state = {queue: [], errors: [], processing: null}, action) {
   switch (action.type) {
     case INITIALIZE:
       return {
         queue: action.payload.queue,

```

Comment:
If anything, since we only have one property now, we should probably get rid of that extra layer (`state.queue.queue`) and just return the queue itself (`state.queue`).